chalk-engine/src/logic.rs

```diff
@@ -913,22 +907,80 @@ impl<C: Context> Forest<C> {
     /// from the NFTD paper.
     fn pursue_positive_subgoal(
```

Comment:
So, these two functions (this and `pursue_negative_subgoal` are really "pursuing" the strand, so a different name might be better. Ideas?